projects/ADuCM3029_demo_cn0397/src/cn0397_app.cpp

```diff
@@ -0,0 +1,519 @@
+    eResult = adi_ble_Init(ApplicationCallback, NULL);
+    DEBUG_RESULT("Error initializing the radio.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    /* Configure radio */
+    eResult = adi_radio_RegisterDevice(ADI_BLE_ROLE_PERIPHERAL);
+    DEBUG_RESULT("Error registering the radio.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    eResult = adi_radio_SetLocalBluetoothDevName(aDataExchangeName, strlen((const char *) aDataExchangeName), 0u, 0u);
+    DEBUG_RESULT("Error setting local device name.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    SetAdvertisingMode();
+
+    /* Initialize data exchange profile */
+    eResult = adi_radio_Register_DataExchangeServer();
+    DEBUG_RESULT("Error registering data exchange server.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    /* Now enter infinite loop waiting for connection and then data exchange events */
+    DEBUG_MESSAGE("Waiting for connection. Initiate connection on central device please.\r\n");
+}
+
+/*!
+ * @brief      Bluetooth visible light demo.
+ *
+ * @details    Bluetooth mode sends the red, green and blue light intensity values to the Android or
+ *             ios application.
+ */
+void VisibleLightBluetoothMode(VisibleLight *pLight)
+{
+    ADI_BLER_RESULT eResult;
+    ADI_BLER_CONN_INFO sConnInfo;
+
+    /* Initialize Bluetooth */
+    InitBluetoothLowEnergy();
+
+    /* WHILE(forever) */
+    while(1u) {
+
+        /* Dispatch events for two seconds - they will arrive in the application callback */
+        eResult = adi_ble_DispatchEvents(ADI_APP_DISPATCH_TIMEOUT);
+        DEBUG_RESULT("Error dispatching events to the callback.\r\n", eResult, ADI_BLER_SUCCESS);
+
+        /* If connected, send data */
+        if(gbConnected == true) {
+            /* Get the connection info. The connection handle will be used to identify which remote device to send data to. */
+            adi_ble_GetConnectionInfo(&sConnInfo);
+
+            if(reg_flag == 0)	{
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(reg_pkt),(uint8_t*)&reg_pkt);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt0),(uint8_t*)&name_pkt0);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt1),(uint8_t*)&name_pkt1);
+
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(name_pkt2),(uint8_t*)&name_pkt2);
+            }
+
+            /* Get the red, green and blue visual light data */
+            pLight->getLightIntensity((float *)&data_pkt.Sensor_Data1);
+
+            if(reg_flag == 1)	{
+            	/* Send the data packet to the remote device */
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            									sizeof(data_pkt),(uint8_t*)&data_pkt);
+            	 DEBUG_MESSAGE("Error sending the data.\r\n");
+            }
+        }
+        /* If disconnected, switch to advertising mode */
+        else
```

Comment:
nitpick: brackets here are not linuxy enough. They should be:
if() {
} else {
}